### PR TITLE
MAKE-1094, MAKE-1102: instrument Okta requests and validate user tokens 

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-1094
https://jira.mongodb.org/browse/MAKE-1102

This is a hodge-podge of random things that have to be done, including token validation. Sorry for the laundry list.

* Verify ID token. Using the Okta l ibrary isn't the most fastest way to verify it but it involves the least code.
* Verify access token. This _should_ be done with the Okta library, but for some reason, the token fails to verify. However, if you send it to the /introspect endpoint, it says the token is valid. I'm going to keep it as is until it actually becomes necessary to fix.
* Add flag to optionally skip populating groups when logging them in. It seems like evergreen ignores these groups. Since it requires a few Okta requests to get the groups, the flag can save some time during authentication.
* Add function that will help match existing LDAP usernames with Okta emails.
* Force the login redirect to show a login prompt to verify your Okta password. This avoids the problem of a logged out user being stealth logged in when they click on a link that requires auth and they're logged out.
* Inject methods to get/put HTTP clients.
* Instrument Okta requests.
* Store and redirect user to page they originally requested before being prompted for login.
* Minor refactors for the redirect handler and redirect callback.

### CODE REVIEW

File: okta/okta.go [hunk 1]
```diff
@@ -468,67 +445,226 @@ func (m *userManager) GetGroupsForUser(user string) ([]string, error) {
...
+	client := m.getHTTPClient()
+	defer m.putHTTPClient(client)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	start := time.Now()
+	resp, err := client.Do(req)
+	grip.Info(message.Fields{
+		"endpoint":        "token",
+		"duration_millis": time.Since(start).Milliseconds(),
```
Comment (john-m-liu) [inline okta/okta.go]: This doesn't compile. But strangely enough the task passes (test fails though)

File: okta/okta.go [hunk 2]
```diff
@@ -260,182 +283,136 @@ func (m *userManager) GetLoginCallbackHandler() http.HandlerFunc {
...
+	}
+	return makeUserFromInfo(userInfo, tokens.AccessToken, tokens.RefreshToken, m.reconciliateID), nil
+}
+
+// generateUserFromIDToken creates a user based on claims in their ID token.
+func (m *userManager) generateUserFromIDToken(tokens *tokenResponse, idToken *jwtverifier.Jwt) (gimlet.User, error) {
+	user, err := makeUserFromIDToken(idToken, tokens.AccessToken, tokens.RefreshToken, m.reconciliateID)
+	if err != nil {
+		err = errors.Wrap(err, "could not generate user from user info received from Okta")
+		grip.Error(err)
```
Comment (john-m-liu) [inline okta/okta.go]: It's probably unnecessary to both log and return the error in these places since the caller will probably log them anyway

Comment (Kimchelly) [reply okta/okta.go]: Removed.

File: okta/okta.go (hunk 1 above)
Comment (Kimchelly) [reply okta/okta.go]: Sorry, this was a go1.13 feature. I fixed it. I also fixed the makefile, which is still probably full of errors but probably fewer of them.

**Pull request merged**